docs/extensibility/create-extension.md

```diff
@@ -0,0 +1,363 @@
+
+    To do so we need to use [the resource_create action](./actions.md#resource_create) and define its configuration under the `with` field:
+
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: my-extension
+      labels:
+        kyma-cli/extension: commands
+        kyma-cli/extension-version: v1
+    data:
+      kyma-commands.yaml: |-
+        metadata:
+          name: configmap
+          description: "Manage ConfigMap resources"
+          descriptionLong: "Use this command to manage ConfigMap resources."
+        subCommands:
+        - metadata:
+            name: create
+            description: "Create ConfigMap resource"
+            descriptionLong: "Use this command to create ConfigMap resource."
+          uses: resource_create
+          with:
+            resource:
+              apiVersion: v1
+              kind: ConfigMap
+              metadata:
+                name: cm-from-extension
+                namespace: default
+    ```
+
+    We can apply new extension version and check if it works:
+
+    ```bash
+    $ kyma alpha configmap create
+
+    resource cm-from-extension applied
+    ```
+
+3. Extend the `create` command with resource-oriented features
+
+    It would be much easier for an end-user to create ConfigMap with defined name, namespace and data. In this section we will use [flags and args](./inputs.md#arguments-and-flags) to collect these data from user and pass them to the `resource_create` action using configuration under the `with` field using the [go tempaltes](./inputs.md#go-templates) and it's available [custom functions](./actions.md#custom-functions):
+
+    ```yaml
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: my-extension
+      labels:
+        kyma-cli/extension: commands
+        kyma-cli/extension-version: v1
+    data:
+      kyma-commands.yaml: |-
+        metadata:
+          name: configmap
+          description: "Manage ConfigMap resources"
+          descriptionLong: "Use this command to manage ConfigMap resources."
+        subCommands:
+        - metadata:
+            name: create <resource_name> [flags]
+            description: "Create ConfigMap resource"
+            descriptionLong: "Use this command to create ConfigMap resource."
+          uses: resource_create
+          args:
+            type: string
+          flags:
+          - name: "namespace"
+            description: "ConfigMap's namespace"
+            shorthand: "n"
+            type: string
+            default: "default"
+          - name: "from-literal"
+            description: "Data element in format <KEY>=<VALUE>"
+            type: map
+          with:
+            resource:
+              apiVersion: v1
+              kind: ConfigMap
+              metadata:
+                name: ${{ .args.value }}
+                namespace: ${{ .flags.namespace.value }}
+              data: ${{ .flags.fromliteral.value | toYaml }}
```

Comment:
Do we have `toYaml` function?

Reply:
We will have
https://github.com/kyma-project/cli/pull/2478